File: templates/Enterprise-Template/src/typescript/generator-botbuilder-enterprise/generators/app/templates/enterprise-bot/src/dialogs/authentication/authenticationResponses.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License

import * as i18n from 'i18n';
import {
    DictionaryRenderer,
    LanguageTemplateDictionary,
    TemplateFunction } from '../templateManager/dictionaryRenderer';
import { TemplateManager } from '../templateManager/templateManager';

export class AuthenticationResponses extends TemplateManager {

    // Fields
    public static responseIds: {
        LoginPrompt: string;
        SucceededMessage: string;
        FailedMessage: string;
    } = {
        LoginPrompt:  'loginPrompt',
        SucceededMessage: 'succeededMessage',
        FailedMessage: 'failedMessage'
    };

    private static readonly responseTemplates: LanguageTemplateDictionary = new Map([
        ['default', new Map([
            [AuthenticationResponses.responseIds.LoginPrompt, AuthenticationResponses.fromResources('authentication.prompt')],
            // tslint:disable-next-line:no-any
            [AuthenticationResponses.responseIds.SucceededMessage, async (data: any): Promise<string> => {
                const value: string = i18n.__('authentication.succeeded');

                return value.replace('{0}', data.name);
            }],
            [AuthenticationResponses.responseIds.FailedMessage, AuthenticationResponses.fromResources('authentication.failed')]
        ])]
    ]);

    constructor() {
        super();
        this.register(new DictionaryRenderer(AuthenticationResponses.responseTemplates));
    }

    private static fromResources(name: string): TemplateFunction {
        return (): Promise<string> => Promise.resolve(i18n.__(name));
    }
}

File: templates/Enterprise-Template/src/csharp/EnterpriseBotSample/EnterpriseBotSample/Dialogs/Authentication/AuthenticationResponses.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using EnterpriseBotSample.Dialogs.Authentication.Resources;
using Microsoft.Bot.Builder;
using Microsoft.Bot.Builder.TemplateManager;
using Microsoft.Bot.Schema;

namespace EnterpriseBotSample.Dialogs.Shared
{
    public class AuthenticationResponses : TemplateManager
    {
        private static LanguageTemplateDictionary _responseTemplates = new LanguageTemplateDictionary
        {
            ["default"] = new TemplateIdMap
            {
                { ResponseIds.LoginPrompt,
                    (context, data) =>
                    MessageFactory.Text(
                        text: AuthenticationStrings.PROMPT,
                        ssml: AuthenticationStrings.PROMPT,
                        inputHint: InputHints.AcceptingInput)
                },
                { ResponseIds.SucceededMessage,
                    (context, data) =>
                    MessageFactory.Text(
                        text: string.Format(AuthenticationStrings.SUCCEEDED, data.name),
                        ssml: string.Format(AuthenticationStrings.SUCCEEDED, data.name),
                        inputHint: InputHints.IgnoringInput)
                },
                { ResponseIds.FailedMessage,
                    (context, data) =>
                    MessageFactory.Text(
                        text: AuthenticationStrings.FAILED,
                        ssml: AuthenticationStrings.FAILED,
                        inputHint: InputHints.IgnoringInput)
                },
            }
        };

        public AuthenticationResponses()
        {
            Register(new DictionaryRenderer(_responseTemplates));
        }

        public class ResponseIds
        {
            public const string LoginPrompt = "loginPrompt";
            public const string SucceededMessage = "succeededMessage";
            public const string FailedMessage = "failedMessage";
        }
    }
}
